File: app/src/main/java/it/corelab/airbooks/adapters/CountriesListAdapter.java
package it.corelab.airbooks.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.Locale;
import java.util.Objects;

import it.corelab.airbooks.R;
import it.corelab.airbooks.section.SignUp;
import it.corelab.airbooks.section.login.fragment.SignUpFragment;


public class CountriesListAdapter extends ArrayAdapter<String> {
    private final Context context;
    private final String[] values;

    public CountriesListAdapter(Context context, String[] values) {
        super(context, R.layout.country_list_item, values);
        this.context = context;
        this.values = values;
    }

    @Override
    public View getView(int position, final View convertView, @NonNull ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        final SignUpFragment sign = new SignUpFragment();

        final Intent adpterIntent = new Intent(context, SignUp.class);

        final View rowView = inflater.inflate(R.layout.country_list_item, parent, false);
        final TextView textView = rowView.findViewById(R.id.txtViewCountryName);

        String[] g=values[position].split(",");
        textView.setText(GetCountryZipCode(g[1]).trim());

        rowView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sign.nation.setText(textView.getText().toString());
                Objects.requireNonNull(SignUpFragment.Companion.getCountryDialog()).dismiss();
            }
        });
        return rowView;
    }

    private String GetCountryZipCode(String ssid){
        Locale loc = new Locale("", ssid);

        return loc.getDisplayCountry().trim();
    }
}

File: app/src/main/java/it/corelab/studios/airbooks/adapters/CountriesListAdapter.kt
package it.corelab.studios.airbooks.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import it.corelab.studios.airbooks.CountryDialog

import java.util.Locale
import java.util.Objects

import it.corelab.studios.airbooks.R
import it.corelab.studios.airbooks.section.login.fragment.SignUpFragment


class CountriesListAdapter(contexts: Context, private val values: Array<String>) : ArrayAdapter<String>(contexts, R.layout.country_list_item, values) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        val rowView = inflater.inflate(R.layout.country_list_item, parent, false)
        val textView = rowView.findViewById<TextView>(R.id.txtViewCountryName)

        val g = values[position].split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        textView.text = getCountryZipCode(g[1]).trim { it <= ' ' }

        rowView.setOnClickListener {
            SignUpFragment.nation.setText(textView.text.toString())
            Objects.requireNonNull<CountryDialog>(SignUpFragment.countryDialog).dismiss()
        }

        return rowView
    }

    private fun getCountryZipCode(ssid: String): String {
        val loc = Locale("", ssid)

        return loc.displayCountry.trim { it <= ' ' }
    }
}